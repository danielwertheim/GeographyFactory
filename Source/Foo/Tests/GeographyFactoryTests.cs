using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Foo.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GeographyFactoryTests
    {
        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_starting_south_to_north_and_many_to_west_and_back_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongSouthToNorthAndManyWestAndBackClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Castle()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.Castle());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Star_drawn_from_middle_and_clockwise()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.StartThatDoesNotRenderWithoutFix());

            Approvals.Verify(g);
        }
    }
}