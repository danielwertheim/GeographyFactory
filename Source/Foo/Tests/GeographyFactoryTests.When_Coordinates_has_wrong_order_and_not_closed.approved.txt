POLYGON ((17.5189979955459 59.8298203729009, 17.5190071588812 59.8296936773323, 17.519133203852 59.8297423369731, 17.5191345140461 59.8298223425544, 17.5189979955459 59.8298203729009))